plugins/modules/syslog_server.py

```diff
@@ -0,0 +1,320 @@
+        host on the HyperCore API will be updated.
+  port:
+    type: int
+    default: 514
+    description:
+      - Network port of the syslog server.
+  protocol:
+    type: str
+    default: udp
+    choices: [ udp, tcp ]
+    description:
+      -  Network protocol used to send syslog alerts.
+  state:
+    type: str
+    choices: [ present, absent ]
+    description:
+      - The desired state of the syslog server on HyperCore API.
+      - If I(state=present) and C(host_new) wasn't provided,
+        a new Syslog server will be added on the HyperCore API.
+      - If I(state=absent), the Syslog server with the provided
+        C(host) will be removed from HyperCore API.
+    required: True
+notes:
+ - C(check_mode) is not supported.
+"""
+
+
+EXAMPLES = r"""
+- name: Create Syslog server
+  scale_computing.hypercore.syslog_server:
+    host: 10.5.11.222
+    port: 514
+    protocol: udp
+    state: present
+
+- name: Update existing Syslog server
+  scale_computing.hypercore.syslog_server:
+    host: 10.5.11.222
+    host_new: 1.2.3.4
+    port: 514
+    protocol: udp
+    state: present
+
+- name: Delete Syslog server
+  scale_computing.hypercore.syslog_server:
+    host: 10.5.11.222
+    state: absent
+"""
+
+RETURN = r"""
+records:
+  description:
+    - Output from modifying entries of the Syslog Servers configuration on HyperCore API.
+  returned: success
+  type: dict
+  sample:
+    alert_tag_uuid: 0
+    host: 10.5.11.222
+    latest_task_tag:
+      completed: 1623069193
+      created: 1623069187
+      descriptionParameters: []
+      formattedDescription: Create Alert Syslog Target
+      formattedMessage: ""
+      messageParameters: []
+      modified: 1623069193
+      nodeUUIDs:
+        - 32c5012d-7d7b-49b4-9201-70e02b0d8758
+      objectUUID: 21c65667-234a-437b-aead-df0199598ff9
+      progressPercent: 100
+      sessionID: ""
+      state: COMPLETE
+      taskTag: 13
+    port: 514
+    protocol: SYSLOG_PROTOCOL_UDP
+    resend_delay: 86400
+    silent_period: 900
+    uuid: 21c65667-234a-437b-aead-df0199598ff9
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ..module_utils import arguments, errors
+from ..module_utils.client import Client
+from ..module_utils.rest_client import RestClient
+from ..module_utils.syslog_server import SyslogServer
+
+UDP = "SYSLOG_PROTOCOL_UDP"  # default
+TCP = "SYSLOG_PROTOCOL_TCP"
+DEFAULT_PORT = 514
+
+
+def get_protocol(protocol: str):
+    return UDP if protocol == "udp" else TCP
+
+
+def create_syslog_server(module: AnsibleModule, rest_client: RestClient):
+    syslog_server = SyslogServer.get_by_host(
+        host=module.params["host"], rest_client=rest_client
+    )
+
+    protocol = get_protocol(module.params["protocol"])
+
+    # If that syslog server already exists, it will not be created again (no duplicates)
+    if syslog_server:
+        before = syslog_server.to_ansible()
+        return False, before, dict(before=before, after=before)
```

Comment:
It would be a bug if this happens. Returning `changed=False` would be incorrect, because `update_syslog_server` should be called, not `create_syslog_server`. Do not return `changed=False`.

You can just `assert syslog_server`.